### DISCUSSION TITLE
Custom-ui strategy going forward

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [Mariusthvdb/custom-ui#128] [Announcements]
Custom-ui is legacy software and more and more so I am afraid. It still works, and is very powerful, but it seems to have no real future in modern HA.

Part of the functionality, the UpdateMoreInfo, was way to fragile to keep working and especially maintain. That's why we now have made the new plugin for that: custom-more-info.

That new plug is modern, and fully compatible with current Dashboard/Frontend, and fully maintainable. @elchininet did a wonderful job there (were still optimizing its options, but even in its current state, it is awesome already)

Because of that I am convinced we should take out at least that functionality of custom-ui, so the 2 don't interfere, and custom-ui has less impact. Custom-ui is updating constantly, and we should prevent that as much as we can.

Next step is to only inject the icon_color attribute, and take out the templating functionality, because that too is updating constantly, upon each and every state change.

A separate plugin custom-icon-color does exactly that, and nothing more. We've even managed to make it work with entity cards now too, so its gotten better than its original :-) (though I presume we can bring that same functionality to custom-ui too, should be easy)

all in all:

## custom-ui (needs to be archived as is)
original, but legacy, offers templating and icon_color, and more-info hiding attributes, with ugly animated dropdown box

## custom-ui-only
original, but legacy, offers templating and icon_color. No more-info (refer to new plugin custom-more-info). This should be the plugin to (try to) maintain in this repo. If possible modernize.

## [custom-icon-color](https://github.com/Mariusthvdb/custom-icon-color/)
original (and yes still legacy), no more templating, offers only icon_color. much lighter and can be used in modern `template:` entities so I can see some future for this small plugin because of that only.

deleted from this repo

___

## [custom-more-info ](https://github.com/Mariusthvdb/custom-more-info/)
new, modern, offers superior customizing of the more-info panels, hides attributes dropdown completely, super nice.

## [custom-attributes](https://github.com/Mariusthvdb/custom-attributes/)
we have also made this plugin, and it handles only the attributes in the more-info panels. It is superseded by custom-more-info, but if you don't want/need the more-info handling of history and logbook, you can keep it as small as this plugin.

**Comment by jonwaland** [comment]
(@Mariusthvdb  Custom-attributes link is broken)
